frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -684,40 +686,53 @@ final class BloopBspServices(
     }
   }
 
+  def jvmRunEnvironment(
+      params: bsp.JvmRunEnvironmentParams
+  ): BspEndpointResponse[bsp.JvmRunEnvironmentResult] =
+    jvmEnvironment(params.targets).map(_.map(new bsp.JvmRunEnvironmentResult(_)))
+
   def jvmTestEnvironment(
       params: bsp.JvmTestEnvironmentParams
-  ): BspEndpointResponse[bsp.JvmTestEnvironmentResult] = {
+  ): BspEndpointResponse[bsp.JvmTestEnvironmentResult] =
+    jvmEnvironment(params.targets).map(_.map(new bsp.JvmTestEnvironmentResult(_)))
+
+  def jvmEnvironment(
+      targets: Seq[BuildTargetIdentifier]
+  ): BspEndpointResponse[List[bsp.JvmEnvironmentItem]] = {
+    def jvmEnvironment(
```

Comment:
nit: should this method be inlined like it was implemented before? It's a bit difficult to read a nested method with the same name as the enclosing method